File: exercises/small_problems/easy5/letter_swap.rb
def swap(str)
  string_array = str.split

  string_array.map! do |str|
    str_arr = str.split("")
    str_arr.unshift(str_arr.pop).push(str_arr[1]).delete_at(1).join
  end

  string_array.join(" ")
end


p swap('Oh what a wonderful day it is') == 'hO thaw a londerfuw yad ti si'
p swap('Abcde') == 'ebcdA'
p swap('a') == 'a'

File: exercises/jsSmallProblems/easy3/letterSwap.js
const letterSwap = (string) => (
  string.slice(-1) + string.slice(1, -1) + string[0]
);

const swap = (string) => string.split(' ')
  .map((word) => (word.length === 1 ? word : letterSwap(word))).join(' ');

console.log(swap('Oh what a wonderful day it is'));// "hO thaw a londerfuw yad ti si"
console.log(swap('Abcde')); // "ebcdA"
console.log(swap('a')); // "a"
